src/plugins/navigationbar.ts

```diff
@@ -0,0 +1,34 @@
+import { Cordova, Plugin } from './plugin';
+
+
+/**
+ * @name NavigationBar
+ * @description
+ * The NavigationBar plugin can you to hide and auto hide the android navigation bar. 
+ *
+ * @usage
+ * ```typescript
+ * import { NavigationBar } from 'ionic-native';
+ *
+ * let autoHide: boolean = true;
+ * NavigationBar.hide(autoHide);
+ * ```
+ */
+@Plugin({
+  name: 'NavigationBar',
+  plugin: 'cordova-plugin-navigationbar',
+  pluginRef: 'plugins.navigationbar',
+  repo: 'https://github.com/cranberrygame/cordova-plugin-navigationbar',
+  platforms: ['Android']
+})
+export class NavigationBar {
+
+  /**
+   * Hide the navigation bar.
+   * @param autohide {boolean} 
+   * @param callbacks {Object} Optional callbacks
+   */
+  @Cordova()
+  static setUp(autohide?: boolean = false, callbacks?: {success?: Function, failure?: Function}): void { }
```

Comment:
This should return a Promise.

Also the decorator should include the following options:
```
@Cordova({
  callbackStyle: 'object',
  successName: 'success',
  errorName: 'failure'
})
```